### DISCUSSION TITLE
[NEW] API : Mettez en place des quotas

### DISCUSSION THREAD

**Discussion by arnolem** [cnumr/best-practices#325] [Bonnes pratiques] [answered]
Hello ! 👋

Le sujet des API est une question que j'ai régulièrement au niveau de l'éco-conception.
Il y a plein de bonnes pratiques WEB à mettre en place aux niveaux des API : 
 - Compression des flux "JSON" / "XML" / ...
 - Fournir des méthodes pour ne récupérer que le nécessaire (Ex : GraphQL)
 - Limiter la consommation du serveur avec un cache côté backend

Toutefois, j'en propose une spécifique aux API : "La mise en place de quotas".

Beaucoup de projets sur lesquels je passe, consomment des APIs sans aucune stratégie de cache.
A chaque chargement de page, on va appeler plusieurs API avant de rendre la page.

La plupart des projets qui font l'effort de mettre en place du cache, c'est souvent parce qu'ils dépassent leur quota (ou que le service est facturé à la consommation).

En mettant en place des quotas sur vos API (en tant que fournisseur), vous imposez une sobriété aux développeurs qui viennent la consommer. A titre d'exemple, j'ai déjà réduit des projets de 10.000 appels par jour à 1 appel par heure.

Qu'en pensez-vous ?

**Comment by ACTLEM** [comment, marked as answer]
Salut @arnolem,

Je suis d'accord avec toi que la mise en place d'une stratégie de cache pour les API HTTP permet de réduire considérablement le nombre d'appels et donc de requêtes HTTP à partir des pages. Je verrais plus une BP comme "Limiter le nombre d'appels aux API HTTP"
Etant donné les délais (fin de la rédaction des BP pour le 4/12), je te laisse proposer une BP sans attendre forcément un retour d'autres personnes.

J'ai créé l'issue pour tracer: https://github.com/cnumr/best-practices/issues/326

**Reply by arnolem** [reply]
Parfait ! 
J'ai bien ta proposition de titre car ça peut se lire dans les deux sens (consommateur et fournisseur)
